### PR TITLE
Add tests for challenge aws utils start workers function

### PR DESCRIPTION
**Pull request by nsjcorps**
I added the tests for challenges aws utils. I got stuck especially with the setup. I used https://github.com/Cloud-CV/EvalAI/pull/2413/files as stated in the task as a base. I created a file in tests/unit/challenges/ folder for the tests. I called it test_aws_utils.py

### CODE REVIEW

**Comment by nsjcorps** [comment]
@Ram81 @yashdusing  @RhishabhJain2018 I am having some issues here. Could you please take a look at the build log

**Comment by nsjcorps** [comment]
@Ram81 @yashdusing  @RhishabhJain2018 I am having some issues here. Could you please take a look at the build log

**Comment by nsjcorps** [comment]
I wrote the test cases. However, they can't work because there are issues with the create_service method. Please take a look at the build.

**Comment by nsjcorps** [comment]
Okay. Let me try @Ram81

**Comment by nsjcorps** [comment]
@Ram81 I have been able to write the other tests. However, for the last one, it asks for a task definition that I don't know what to fill in.

**Comment by nsjcorps** [comment]
Thank you for your help. I must say this was a hard one @Ram81  but it's good for the experience

**Comment by nsjcorps** [comment]
@Ram81 I have been able to complete the tests. Apparently the task_definition was just the queue_name which I have filled in

**Comment by nsjcorps** [comment]
Please take a look @Ram81 @yashdusing @KhalidRmb @RishabhJain2018 @pushkalkatara

**Comment by nsjcorps** [comment]
I got more inspiration from the AWS website. I read some of the docs there.  @Ram81 . The link you gave me was more on requests but the problem was with the position of the deployment configuration and the cluster name. I got the real position from the documentation

**Comment by pushkalkatara** [comment]
I think still several tests are left from `aws_utils.py`, but this PR looks a lot of work by the student @Ram81 Should I split it into multiple different GCI tasks for unit/integration tests?

**Comment by nsjcorps** [comment]
@pushkalkatara. I left you a message on Gutter sir

**Comment by nsjcorps** [comment]
@pushkalkatara sorry, the title must have been deceiving. It was just for the start workers function

File: tests/unit/challenges/test_aws_utils.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
+import boto3
+import challenges.aws_utils as aws_utils
```
Comment (vkartik97) [inline tests/unit/challenges/test_aws_utils.py]: Please move this down after line 6

Comment (nsjcorps) [reply tests/unit/challenges/test_aws_utils.py]: Done

**Comment by nsjcorps** [comment]
@KhalidRMB I have done that. Thanks

**Comment by nsjcorps** [comment]
Thanks! @KhalidRmb

**Comment by yashdusing** [comment]
you don't need to remove deploymentController. The problem you've been experiencing is either due to the use of '{{' or improper unpacking while **VPC (newline characters and being in included as key). You could simply call do
`
definition['deploymentController'] = {
            "type": "ECS"
}
`
after eval has been called. I've tested on local and it worked

**Comment by yashdusing** [comment]
@nsjcorps you can refer #2604 for solving the deploymentController issue.

**Comment by nsjcorps** [comment]
@yashdusing I am working on it

**Comment by yashdusing** [comment]
also, change the order of installation in travis

**Comment by nsjcorps** [comment]
I'm not to sure I understand @yashdusing

**Comment by nsjcorps** [comment]
Oh! I get now @yashdusing

**Comment by nsjcorps** [comment]
I got an error @yashdusing

File: apps/challenges/aws_utils.py [hunk 2]
```diff
@@ -221,10 +221,10 @@
             "assignPublicIp": "ENABLED"
         }}
     }},
-    "schedulingStrategy":"REPLICA",
-    "deploymentController":{{
-        "type": "ECS"
+    "deploymentController": {{
+        "type": "ECS
```
Comment (yashdusing) [inline apps/challenges/aws_utils.py]: @nsjcorps you forgot the ending "

**Comment by nsjcorps** [comment]
@yashdusing I am still getting an error

**Comment by yashdusing** [comment]
@nsjcorps I guess this error has to do with migrations not being applied. I'm not sure about this one. But, the aws_utils passed so it's probably not related to boto3

**Comment by nsjcorps** [comment]
Alright @yashdusing . Thanks a million . Glad to hear this issue is resolved. @nikochiko Nice Work on your side! You did a great job!

**Comment by yashdusing** [comment]
@nsjcorps Are you sure that the db and migrations are in sync with the master branch ? You can sync the branch with master and check once

**Comment by nsjcorps** [comment]
@yashdusing the branch is already up to date

**Pull request closed without merging**